src/PrestaShopBundle/Controller/Admin/Improve/International/LanguageController.php

```diff
@@ -59,21 +61,43 @@ class LanguageController extends AbstractAdminController
      *
      * @param Request $request
      * @param LanguageFilters $filters
+     * @param CurrencyFilters $currencyFilters
      *
      * @return Response
      */
-    public function indexAction(Request $request, LanguageFilters $filters)
+    public function indexAction(Request $request, LanguageFilters $filters, CurrencyFilters $currencyFilters)
```

Comment:
it's a BC except if this migration has been introduced in 1.7.6

Reply:
So you're saying once a controller is released it can never change its signature? have new parameters?
What do you suggest? should we have the new parameter optional? or should be create a new action and leave the former one even f it's not used any more? Deprecate the former fonction, or deprecate the whole controller?
Or maybe we should define our controller classes as `final` ? This way we should be allowed to modify them no?

Anyway this controller is new in 176 so we are good for this one, but we are going to face these kind of issues sooner or later.

Reply:
I'm not sure about controller actions 🤔 because:
- they cannot be overriden
- another php class will never call `$controller->indexAction($request, $filters)` , it will always be called by Symfony

Reply:
That's what I thought as well, and we can define them as final classes to avoid them being inherited

Reply:
Also what we're discussing here is just an exemple, in the final PR it will be gone 😋

Reply:
Oh I thought it was needed! But ok, anyway it works fine, filters in one page are saved and displayed in the other